### DISCUSSION TITLE
hope upload component support custom list render

### DISCUSSION THREAD

**Discussion by DophinL** [ant-design/ant-design#44199] [Ideas]
what i need is virtual render upload list, which improve performance when there a tons of files to be upload.

**Comment by zombieJ** [comment]
Just disable the list and apply custom one. Upload is not strong binding with List.

**Reply by DophinL** [reply]
thx, you means `showUploadList: false` and render by fileList in my own way right?

**Comment by kiner-tang** [comment]
Maybe you can try to use `itemRender` to customize your upload list.
![image](https://github.com/ant-design/ant-design/assets/10286961/c24349b1-13a7-4d2d-b86b-fc37e249aaa3)
